src/config.rs

```diff
@@ -41,3 +97,103 @@ pub mod mock {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::fs;
+    use std::fs::File;
+    use std::path::PathBuf;
+    use std::io::prelude::*;
+    use super::{Config, FsConfig};
+
+    fn clean_up(test_root: &PathBuf) {
+        if test_root.exists() {
+            fs::remove_dir_all(test_root).unwrap();
+        }
+        assert!(!test_root.is_dir());
+    }
+
+    fn set_up(suffix: &str, current: &str, shells: Vec<&str>) -> PathBuf {
+        let test_root = PathBuf::from("./target/fs-config-tests-".to_owned() + suffix);
+
+        clean_up(&test_root);
+        fs::create_dir(&test_root).unwrap();
+        assert!(test_root.is_dir());
+
+        let path = test_root.join("current_shell");
+        let mut file = File::create(&path).unwrap();
+        file.write_all(current.as_bytes()).unwrap();
+
+        let shell_root = test_root.join("shells");
+        fs::create_dir(&shell_root).unwrap();
+        for shell in shells {
+            let new_shell = shell_root.join(PathBuf::from(shell));
+            fs::create_dir(&new_shell).unwrap();
+        }
+
+        test_root
+    }
+
+    #[test]
+    fn has_a_root_path() {
+        let test_root = set_up("root-path", "default", vec!["default"]);
+        let config = FsConfig::new(test_root.clone()).unwrap();
+        assert_eq!(config.root_path(), &test_root);
+
+        clean_up(&test_root);
+    }
+
+    #[test]
+    fn returns_the_current_shell_name() {
+        let test_root = set_up("current-shell-name", "current", vec!["default"]);
+        let config = FsConfig::new(test_root.clone()).unwrap();
+        assert_eq!(config.current_shell_name(), "current".to_string());
+
+        clean_up(&test_root);
+    }
+
+    #[test]
+    fn can_set_the_current_shell_name() {
+        let test_root = set_up("set-current-shell-name", "default", vec!["default"]);
+        let mut config = FsConfig::new(test_root.clone()).unwrap();
+        config.set_current_shell_name("current").unwrap();
+
+        let mut config_file = File::open(&test_root.join("current_shell")).unwrap();
+        let mut name_on_disk = String::new();
+        config_file.read_to_string(&mut name_on_disk).unwrap();
+
+        let current = "current".to_string();
+        assert_eq!(config.current_shell_name(), current);
+        assert_eq!(name_on_disk, current);
+
+        clean_up(&test_root);
+    }
+
+    #[test]
+    fn can_confirm_a_shell_exists() {
+        let test_root = set_up("confirm-shell-existence",
+                               "default",
+                               vec!["default", "other"]);
+        let config = FsConfig::new(test_root.clone()).unwrap();
+
+        let shell_exists = config.does_shell_exist("other");
+        assert!(shell_exists);
```

Comment:
Haha. I was actually about to do that and decided against it 'cuz you hadn't asked for it earlier, but I'm down. I'll actually just `--amend` my previous commit.